CMakeLists.txt

```diff
@@ -71,15 +71,19 @@ rocm_setup_version( VERSION ${VERSION_STRING} )
 
 option( BUILD_VERBOSE "Output additional build information" OFF )
 
-option( BUILD_WITH_SOLVER "Add additional functions from rocSOLVER" ON )
-
-if( BUILD_WITH_SOLVER )
-    add_definitions( -D__HIP_PLATFORM_SOLVER__ )
-endif( )
-
 # BUILD_SHARED_LIBS is a cmake built-in; we make it an explicit option such that it shows in cmake-gui
 option( BUILD_SHARED_LIBS "Build hipBLAS as a shared library" ON )
 
+if(BUILD_SHARED_LIBS)
+    set(BUILD_WITH_SOLVER_DEFAULT OFF)
+else()
```

Comment:
Is this just for testing or default to not link?    Default to not link may may need to review effect on rpath or package consumption documentation.

Reply:
Default will be to not link. I think it'll be ok but I'll test out consuming the hipblas package now to make sure it find rocsolver.